### ISSUE TITLE
Failure to register a ServiceWorker from vscode-cdn.net when opening a Jupyter notebook using VS Code

### ISSUE THREAD

**Issue by l0rd** [eclipse/che#21852]
### Describe the bug

When trying to open https://github.com/rh-aiservices-bu/object-detection-rest.git on the dogfooding instance (can be tested on [RH Developer Sandbox](https://workspaces.openshift.com/#https://github.com/rh-aiservices-bu/object-detection-rest?che-editor=https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/insiders/devfile.yaml) too):

**On chrome with normal window it loads successfully 👍**

<img width="1045" alt="image" src="https://user-images.githubusercontent.com/606959/204872565-fe133035-6944-4bdd-afd0-6f9ed043a3d9.png">

**On chrome with incognito window it fails 🐛**
<img width="374" alt="image" src="https://user-images.githubusercontent.com/606959/204873295-cc772565-d337-4ad4-9382-8d388c4a3085.png">

<img width="461" alt="image" src="https://user-images.githubusercontent.com/606959/204872384-ffaf3bb4-6a0a-4863-b093-8e1b786abb51.png">

<img width="1766" alt="image" src="https://user-images.githubusercontent.com/606959/204872221-2b94ffb6-9343-44fe-87f1-a85847e269c3.png">

<img width="892" alt="image" src="https://user-images.githubusercontent.com/606959/204873077-01664f14-b213-4dec-87be-c0bd9c7c195b.png">

**On firefox with normal window it fails 🐛**

<img width="661" alt="image" src="https://user-images.githubusercontent.com/606959/204874248-e07a64a8-a767-40ca-9344-e7711482b131.png">

<img width="554" alt="image" src="https://user-images.githubusercontent.com/606959/204846553-0415823b-8e1c-4ae5-b9f1-27488f414f3a.png">

<img width="1363" alt="image" src="https://user-images.githubusercontent.com/606959/204846677-c5ecdf1e-889d-4f78-8e8e-edcf02e4af76.png">

<img width="1776" alt="image" src="https://user-images.githubusercontent.com/606959/204873465-ea26907a-7a54-4997-aae1-2cfefd16da5f.png">

### Che version

next (development version)

### Steps to reproduce

Click [here](https://workspaces.openshift.com/#https://github.com/rh-aiservices-bu/object-detection-rest?che-editor=https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-code/insiders/devfile.yaml) and open the file `0_sandbox.ipynb`.

### Expected behavior

The notebook should be displayed correctly and it should be possible to execute all cells (the notebook is rendered as a blank page instead). 

### Runtime

OpenShift

### Installation method

OperatorHub

### Environment

Amazon

### Additional context

The version of che-code I have tested against is `1.74.0`

<img width="518" alt="image" src="https://user-images.githubusercontent.com/606959/205026244-dc0bf3e7-6e75-4b28-bc52-1b8a86c1a436.png">

**Comment by azatsarynnyy** [comment]
I verified it in my environment which is Fedora Linux 36. I can reproduce it in Chrome's Incognito mode only.
It seems the issue is reproduced in Firefox on Mac, but not in Linux. Similar as https://github.com/eclipse/che/issues/21782

**Comment by azatsarynnyy** [comment]
Below :arrow_down: are the results of my quick research on this topic.

The problem is caused by the browser's mechanism of cookie blocking. Since they turn it on in Standart (non-Incognito) mode this year, the VS Code users started facing unexpected problems with different parts of functionality. Here are some of the cases:
- https://github.com/microsoft/vscode/issues/155033
- https://github.com/microsoft/vscode-pull-request-github/issues/3671#issuecomment-1199015668
- https://github.com/microsoft/vscode/issues/142813
- https://github.com/microsoft/vscode/issues/166131
- https://github.com/microsoft/vscode/issues/143035

As I understand, someone from the VS Code team is already looking at this problem. See https://github.com/microsoft/vscode/issues/166131#issuecomment-1312158666.

While it's still a known issue in VS Code upstream, I found how to work the problem around by updating the browser's settings.

To make it work in Google Chrome Incognito mode:
1. Go to `chrome://settings/cookies`.
3. Choose `Allow all cookies`:
![image](https://user-images.githubusercontent.com/1636395/205022715-a95903c8-e9b2-422e-88a2-742f1a7e52db.png)

I'm not able to check it on Mac but based on the information I found the steps are the following.
To make it work in Firefox Standard mode on Mac (no issues on Linux):
1. Go to `about:preferences#privacy`.
2. Try with the disabled `Test pilot our most powerful privacy feature ever` checkbox.
3. Another option to try is the shield button and turning off the protection:
![image](https://user-images.githubusercontent.com/1636395/205021407-bed28813-69c8-4e32-b6cb-66d75c92a08b.png)

It's not possible to make it work in Firefox Private window as Firefox doesn't support the ServiceWorkers at all when in Private Browsing Mode. ServiceWorkers are the underlying technology on which WebView is built. See the Mozilla issue for more details: https://bugzilla.mozilla.org/show_bug.cgi?id=1320796

**Comment by l0rd** [comment]
Thank you @azatsarynnyy for this analysis!

For chrome incognito I have verfied that it works on mac too and I have figured out that we can also allow cookies for the site `vscode-cdn.net` only:

<img width="668" alt="image" src="https://user-images.githubusercontent.com/606959/205030203-8b5f7f10-e6ad-4d9e-a2e8-5d7cf051594f.png">

In general we should mention this workaround in documentation or known issues. Otherwise users always get an error when loading VS Code in firefox or chrome incognito (not just when using Jupyter).
